lib/routes/account.js

```diff
@@ -1434,6 +1444,80 @@ module.exports = function (
     },
     {
       method: 'POST',
+      path: '/account/login/send_unblock_code',
+      config: {
+        validate: {
+          payload: {
+            email: validators.email().required()
+          }
+        }
+      },
+      handler: function (request, reply) {
+        log.begin('Account.SendUnblockCode', request)
+
+        var email = request.payload.email
+        var emailRecord
+        var ip = request.app.clientAddress
+
+        return P.resolve()/*customs.check(request, email, 'sendUnblockCode')*/
+          .then(lookupAccount)
+          .then(createUnblockCode)
+          .then(mailUnblockCode)
+          .done(function () {
+            reply({})
+          }, reply)
+
+        function lookupAccount() {
+          return db.emailRecord(email)
+            .then(function(record) {
+              emailRecord = record
+              return record.uid
+            })
+        }
+
+        function createUnblockCode(uid) {
+          return db.createUnblockCode(uid)
+        }
+
+        function mailUnblockCode(code) {
+          return getGeoData(ip)
+            .then(function (geoData) {
+              return mailer.sendUnblockCode(emailRecord, code, userAgent.call({
```

Comment:
Pass the whole emailRecord to the auth-mailer here, both the uid and the email are needed.